File: src/lib/format.ts
/** Human, calm date/time formatting for invitations. */

const DAYS = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
const MONTHS = [
  'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
  'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
];

export function formatWhen(iso: string): string {
  const d = new Date(iso);
  return `${DAYS[d.getDay()]}, ${d.getDate()} ${MONTHS[d.getMonth()]} · ${formatTime(d)}`;
}

export function formatTime(d: Date): string {
  const h = d.getHours();
  const m = d.getMinutes().toString().padStart(2, '0');
  return `${h.toString().padStart(2, '0')}:${m}`;
}

export function formatDuration(min: number): string {
  if (min < 60) return `${min} min`;
  const h = min / 60;
  return Number.isInteger(h) ? `${h}h` : `${Math.floor(h)}h ${min % 60}m`;
}

export function endTimeLabel(iso: string, durationMin: number): string {
  const end = new Date(new Date(iso).getTime() + durationMin * 60000);
  return formatTime(end);
}

File: legacy/v2/lib/data/format.dart
/// Human, calm date/time formatting for invitations.
library;

const _days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
const _months = [
  'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
  'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
];

String _two(int n) => n.toString().padLeft(2, '0');

String formatTime(DateTime d) => '${_two(d.hour)}:${_two(d.minute)}';

String formatWhen(DateTime d) {
  final day = _days[d.weekday - 1];
  final month = _months[d.month - 1];
  return '$day, ${d.day} $month · ${formatTime(d)}';
}

String formatDuration(int minutes) {
  if (minutes < 60) return '$minutes min';
  final h = minutes / 60;
  return h == h.roundToDouble() ? '${h.toInt()}h' : '${minutes ~/ 60}h ${minutes % 60}m';
}

String endTimeLabel(DateTime start, int durationMin) {
  return formatTime(start.add(Duration(minutes: durationMin)));
}
